src/cli/self_update.rs

```diff
@@ -444,8 +444,10 @@ fn do_pre_install_options_sanity_checks(opts: &InstallOpts) -> Result<()> {
 // sudo is configured not to change $HOME. Don't let that bogosity happen.
 #[allow(dead_code)]
 fn do_anti_sudo_check(no_prompt: bool) -> Result<()> {
+    use std::ffi::OsString;
+
     #[cfg(unix)]
-    pub fn home_mismatch() -> bool {
+    pub fn home_mismatch() -> (bool, OsString, &'static str) {
```

Comment:
Hmm, is this `'static` real? I'd be slightly worried that whatever is coming out of this function is not really static.

Reply:
I wouldn't think it would be either but the compiler seemed okay with it. I can convert it to a `String` if that's better?

Reply:
I'd feel happier with a String or OsString really yes.